### PR TITLE
fix(manifest): preserve default network config on empty env overrides

### LINKED ISSUE #2420: How to configure security groups in Load Balancer Web Service
Hi, there.
I would like to set my custom security group to ECS tasks, so I add network configuration to `manifest.yml` as documented [here](https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#network-vpc-security-groups).

```yaml
name: rails
type: Load Balanced Web Service
...
network:
  vpc:
    security_groups: ['sg-09ce2df3852b84be9']
```

And deploy these commands.

```sh
$ copilot version
version: v1.7.0, built for darwin

$ copilot svc deploy -e dev -n rails 
```

In AWS Management Console, `sg-09ce2df3852b84be9` is not set to ECS tasks. Only the security group set with Copilot by default is configured for the ECS tasks.

スクリーンショット 2021-06-04 23 19 17

It seemed that the security group set in `manifest.yml` was not set in the CloudFormation template as well.

```sh
$ copilot svc package -e dev -n rails
...
Service:
    Metadata:
      'aws:copilot:description': 'An ECS service to run and maintain your tasks in the environment cluster'
    Type: AWS::ECS::Service
    DependsOn: WaitUntilListenerRuleIsCreated
    Properties:
      Cluster:
        Fn::ImportValue:
          !Sub '${AppName}-${EnvName}-ClusterId'
      TaskDefinition: !Ref TaskDefinition
      DesiredCount: !Ref TaskCount
      DeploymentConfiguration:
        DeploymentCircuitBreaker:
          Enable: true
          Rollback: true
        MinimumHealthyPercent: 100
        MaximumPercent: 200
      PropagateTags: SERVICE
      EnableExecuteCommand: true
      LaunchType: FARGATE
      NetworkConfiguration:
        AwsvpcConfiguration:
          AssignPublicIp: ENABLED
          Subnets:
            - Fn::Select:
              - 0
              - Fn::Split:
                - ','
                - Fn::ImportValue: !Sub '${AppName}-${EnvName}-PublicSubnets'
            - Fn::Select:
              - 1
              - Fn::Split:
                - ','
                - Fn::ImportValue: !Sub '${AppName}-${EnvName}-PublicSubnets'
          SecurityGroups:
            - Fn::ImportValue: !Sub '${AppName}-${EnvName}-EnvironmentSecurityGroup'
            - Fn::GetAtt: [AddonsStack, Outputs.railsclusterSecurityGroup]
...
```

Any suggestion would be appreciated.
Thanks in advance!

### PR DESCRIPTION
**Pull request by efekarakus**
Fixes #2420

Previously, when a user provided a network config and an env override:
```yaml
network:
  vpc:
    security_groups: ['sg-123123']
environments:
  dev:
    count: 2
```
The network config for the "dev" environment got overriden to an empty
struct{}.

This change ensures that the default entry is preserved for network
config is preserved for the "dev" environment.
This is done by changing the type for the Network field to a pointer so
that mergo does not overwrite it.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/manifest/workload.go [hunk 1]
```diff
@@ -438,21 +438,25 @@ type TaskConfig struct {
...
 	conf := networkWithDefaults{
-		VPC: vpcConfig{
-			Placement: stringP(PublicSubnetPlacement),
-		},
+		VPC: defaultVPCConf,
 	}
 	if err := unmarshal(&conf); err != nil {
 		return err
 	}
+	if conf.VPC == nil { // If after unmarshaling the user didnot specify VPC configuration then reset it to public.
```
Comment (huanjani) [inline internal/pkg/manifest/workload.go]: nit:
```suggestion
	if conf.VPC == nil { // If after unmarshaling the user did not specify VPC configuration then reset it to public.
```

Comment (efekarakus) [reply internal/pkg/manifest/workload.go]: fixed the typo!

File: internal/pkg/deploy/cloudformation/stack/transformers.go [hunk 2]
```diff
@@ -533,7 +533,13 @@ func convertEFSConfiguration(in manifest.EFSVolumeConfiguration) *template.EFSVo
...
+func convertNetworkConfig(network *manifest.NetworkConfig) *template.NetworkOpts {
+	if network == nil || network.VPC == nil {
+		return &template.NetworkOpts{
+			AssignPublicIP: template.EnablePublicIP,
+			SubnetsType:    template.PublicSubnetsPlacement,
+		}
+	}
 	opts := &template.NetworkOpts{
 		AssignPublicIP: template.EnablePublicIP,
 		SubnetsType:    template.PublicSubnetsPlacement,
```
Comment (iamhopaul123) [inline internal/pkg/deploy/cloudformation/stack/transformers.go:545]:
```suggestion
	opts := &template.NetworkOpts{
		AssignPublicIP: template.EnablePublicIP,
		SubnetsType:    template.PublicSubnetsPlacement
	}
	if network == nil || network.VPC == nil {
		return opts
	}
	opts.SecurityGroups = network.VPC.SecurityGroups
```

**Review by iamhopaul123 — APPROVED** [review]
:shipit:

**Review by huanjani — APPROVED** [review]
lgtm!

**Pull request merged**